packages/vite/src/node/config.ts

```diff
@@ -416,6 +447,7 @@ export async function resolveConfig(
     assetsInclude(file: string) {
       return DEFAULT_ASSETS_RE.test(file) || assetsFilter(file)
     },
+    knownJsSrcExtensions: resolvedKnownJsSrcExtensions,
```

Comment:
We discussed a bit about this in Discord. IMO we could also use the same scheme utilized by `assetsInclude` here. In the ResolvedConfig, we will have a `resolved.isJsRequest(path)` function that plugins can then use. This is something else to discuss with Evan, if we go here, we may want to have all these helpers in `resolve.utils.isJsRequest`, but I am not sure if it is really needed and the flat structure is enough here. @antfu what do you think?